### ISSUE TITLE
Polling stops when internet connection is lost

### ISSUE THREAD

**Issue by kunago** [xLinkOut/telegram-udf-autoit#14]
Hey, I have to thank you very much for this UDF. It is truly excellent. I am using it to automate thr communication with remote computers.

I currently have one issue though. I run the polling script when user logs in and it should keep polling as long as the user is logged in. I noticed however that once an internet connection is lost, even for a few minutes, the script returns 0x1 code and stops.

I can understand it stops but is there any way for polling to simply skip checking for new messages while a computer is not accessible and start polling again when it is online? I mean some sort of check for the script not to end?

I found this solution https://eddiejackson.net/wp/?p=7530 and I tested another one so far. Well, they don't really seem to do the job.

**Comment by xLinkOut** [comment]
Ehi! Thanks, first of all. I appreciate it!
So basically when there is no internet connection the script close itself because throw an error. I think that this is due to the http functions; can you please tell me at which line the script stop? Maybe it just need some sort of try/catch logic.

**Comment by kunago** [comment]
Hard to tell actually as I am running the script via Windows scheduler. It runs for a while and than stops with the error 0x1. I can take a close look at it though.

I also looked at some "ping" solutions but I cannot really imaging pinging google or any other such website every few seconds. I tried to implement the solution outlined above but it did nothing,

**Comment by xLinkOut** [comment]
Okay, thanks for reporting this. I'll check it out.

**Comment by kunago** [comment]
I found something in the forum that could be helpful: https://www.autoitscript.com/forum/topic/192470-winhttp-request-fail/?do=findComment&comment=1381321

Also see Example 2 on this page: https://www.autoitscript.com/autoit3/docs/functions/ObjEvent.htm

The UDF seems to fail in __HttpGet but when this happens, the script exists. The above initiates a custom error handler that does not exit the script.

Could that be a way?

**Comment by sergeyFlakon** [comment]
Hey guys, I was facing the same issue, what I did was:
Telegram.au3 got additional function added

```
Func _PollingNoLoop()
    While 1
        Sleep(1000) ;Prevent CPU Overloading
        $newUpdates = _GetUpdates()
        ConsoleWrite($newUpdates & @CRLF)
        If Not StringInStr($newUpdates,'update_id') Then Return
        $msgData = __MsgDecode($newUpdates)
        $Offset = $msgData[0] + 1
        ConsoleWrite(_ArrayToString($msgData) & @CRLF)
        Return $msgData
    WEnd
EndFunc ;==> _PollingNoLoop
```

My main bot script had it included and used polling this way 

```
Func _TelegramPolling()
Global	$msgData = _PollingNoLoop()
	_ArrayDisplay($msgData)
If	IsArray($msgData) Then
_TelegramAnalyzeMessage()
EndIf
EndFunc

While 1
   	$msgData = _PollingNoLoop()
If	IsArray($msgData) Then
ConsoleWrite("Received array, message is " & $msgData[5] & @CRLF)
_TelegramAnalyzeMessage()
	EndIf
WEnd
```

**_TelegramAnalyzeMessage()** is a func which analyzed all incoming messages

This was a long time ago, and I'm not using autoit for it anymore so I don't recall exactly why I had NoPolling func added, I think the reason is to handle it within my script, and here's the handle part:
```
If	IsArray($msgData) Then
ConsoleWrite("Received array, message is " & $msgData[5] & @CRLF)
_TelegramAnalyzeMessage()
	EndIf
```

Basically, if $msgData exists, we do continue running func, if not (network outage) - we don't)
Hope this will help you @kunago

**Comment by kunago** [comment]
I tried to look into it but failed.

Using the method I outlined did work for a few trials but then as I found in other forums this "ObjCreate("WinHttp.WinHttpRequest.5.1")" wears out and it fails eventually.

I could not find any meaningful function and did not want to go the "ping" was so I created a "healer" that checks for program exist statuses and once it finds it had failed, it restarts the program.

**Comment by FreiherrKenny** [comment]
Same problem seems to happen when i disconnect from OpenVPN.
````
#Region "@HTTP Request"
Func _Telegram_HttpGet($sURL,$sData = '')
    Local $oHTTP = ObjCreate("WinHttp.WinHttpRequest.5.1")
    $oHTTP.Open("GET",$sURL & "?" & $sData,False)
    If (@error) Then Return SetError(1,0,0)
    $oHTTP.Send()
    If (@error) Then Return SetError(2,0,0)
    If ($oHTTP.Status <> $HTTP_STATUS_OK) Then Return SetError(3,0,0)
    Return SetError(0,0,$oHTTP.ResponseText)
EndFunc ;==> _Telegram_HttpGet
````
```
==> The requested action with this object has failed.:
$oHTTP.Send()
$oHTTP^ ERROR
```

**Comment by FreiherrKenny** [comment]
Error got solved in my fork.
@kunago you might wanna check the last update of my fork to handle this error properly

**Comment by kunago** [comment]
I was unable to solve the issue using this library so I get to give up on it.
